### PR TITLE
container_run_* rules silent on success

### LINKED ISSUE #2006: Stop spewing when running
# 🐞 bug report

### Affected Rule

Not sure

### Is this a regression?

No

### Description

In our logs we see lots of things like
```
(18:46:34) INFO: From RunAndCommitLayer keys/keys/tests/test_py36_pip_pytest-layer.tar:
  | Loaded image: bazel/keys/keys/tests:test_py36_pip_pytest_requirements
  | sha256:d03e104673a2a6db4bba77050f7a2765c8fef90b5a3610615fc263300a16df13
  | 959deff37fc3ab917a744cc772926ff9d41f6f7e45abe8e6b8b5aeac32214164
  | Untagged: bazel/keys/keys/tests:test_py36_pip_pytest
  | Deleted: sha256:d03e104673a2a6db4bba77050f7a2765c8fef90b5a3610615fc263300a16df13
  | Deleted: sha256:9ee379ff3269fde70400605b3853fb8670ef82b887180af2934fd5dfc9d7ecfe
  | a66f73f21e529d31d202471d54e060d2ebe6784687ea7e5e55030244d91447c5
```

But on a successful run of bazel rules they should be silent

## 🔬 Minimal Reproduction

Not sure

## 🔥 Exception or Error

None

## 🌍  Your Environment

**Operating System:**

linux

**Output of `bazel version`:**

```
$ bazel version
Bazelisk version: development
Starting local Bazel server and connecting to it...
Build label: 4.2.0
Build target: bazel-out/darwin-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Wed Aug 18 12:54:03 2021 (1629291243)
Build timestamp: 1629291243
Build timestamp as int: 1629291243
```

**Rules_docker version:**

```
    http_archive(
        name = "io_bazel_rules_docker",
        sha256 = "db427107e4c36a3713c24daf5fcdf7719fe664e0c36e089b5b16d15cf331ce60",
        strip_prefix = "rules_docker-8a4f73fb29a64ba813087220b200f49a1ca10faa",
        # A few commits past the latest 0.22.0 release so that we pick up
        # https://github.com/bazelbuild/rules_docker/commit/8a4f73fb29a64ba813087220b200f49a1ca10faa
        # which is required to support rules_nodejs 5.0
        urls = ["https://github.com/bazelbuild/rules_docker/archive/8a4f73fb29a64ba813087220b200f49a1ca10faa.tar.gz"],
    )
```

**Anything else relevant?**

### PR DESCRIPTION
**Pull request by alexeagle**
Prevents RunAndCommitLayer and related actions from spamming the Bazel stdout log with Docker output, when the action succeeds. Similar to #1969 which did this for ExtractImageId.

Fixes #2006

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by ptarjan** [comment]
Thanks so much for fixing this! What's the process to get it merged?

**Comment by alexeagle** [comment]
Need a review ... the maintainership on this repo is stretched very thin right now. @gravypod any response on your internal email thread about that?

**Comment by ptarjan** [comment]
Thanks. I can verify it does indeed fix all the issues in the linked issue for us.

**Comment by alexeagle** [comment]
Note, @ptarjan reports there are two more cases to fix:

```
$ bazel build bazel/docker/internal/test/image_with_packages_download.tar
INFO: Invocation ID: a91cc674-678a-4099-942e-d2c3ae98ea37
INFO: Analyzed target //bazel/docker/internal/test:image_with_packages_download.tar (66 packages loaded, 10815 targets configured).
INFO: Found 1 target...
INFO: From Action bazel/docker/internal/test/image_with_packages_download.tar:
Loaded image: bazel/bazel/docker/internal/test:ubuntu_image
37f4acdb6845a63e8dce1d1b5a02d0c179c779bb24b2b760e41556ffbfe74788
Target //bazel/docker/internal/test:image_with_packages_download.tar up-to-date:
  bazel-bin/bazel/docker/internal/test/image_with_packages_download.tar
INFO: Elapsed time: 28.350s, Critical Path: 16.55s
INFO: 76 processes: 56 disk cache hit, 20 internal.
INFO: Build completed successfully, 76 total actions
```
which I think is from container_image and

```
$ bazel bazel/docker/internal/test/image_with_packages_install.tar
INFO: Invocation ID: df0a3773-472e-4079-a136-2063ffcd9240
INFO: Analyzed 1 target (1 packages loaded, 9 targets configured).
INFO: Found 1 target...
INFO: From ExtractImageId bazel/docker/internal/test/image_with_packages_install.tar.unstripped:
Loaded image: bazel/bazel/docker/internal/test:ubuntu_image
Target //bazel/docker/internal/test:image_with_packages_install.tar up-to-date:
  bazel-bin/bazel/docker/internal/test/image_with_packages_install.tar
INFO: Elapsed time: 5.167s, Critical Path: 4.65s
INFO: 12 processes: 2 disk cache hit, 10 internal.
INFO: Build completed successfully, 12 total actions
```
which I think is from install_pkgs

File: docker/util/commit.sh.tpl [hunk 1]
```diff
@@ -15,18 +15,24 @@ if [[ -z "$DOCKER" ]]; then
...
-  $DOCKER $DOCKER_FLAGS logs $id && false
-fi
+if ! (
+    # Load the image and remember its name
+    image_id=$(%{image_id_extractor_path} %{image_tar})
+    $DOCKER $DOCKER_FLAGS load -i %{image_tar}
+
+    readonly id=$($DOCKER $DOCKER_FLAGS create %{docker_run_flags} $image_id %{commands})
+    retcode=0
+    if docker start -a "${id}"; then
```
Comment (mattem) [inline docker/util/commit.sh.tpl:27]:
```suggestion
    if $DOCKER $DOCKER_FLAGS start -a "${id}"; then
```

**Review by mattem — CHANGES_REQUESTED** [review]

**Review by mattem — APPROVED** [review]

**Comment by alexeagle** [comment]
replaced by #2047

**Pull request closed without merging**